frontend/catalyst/autograph.py

```diff
@@ -197,20 +198,14 @@ def else_body():
     )
 
 
-def disable_autograph(fn):
-    """Decorator that disables AutoGraph for the given function.
+class disable_autograph(ag_ctx.ControlStatusCtx, ContextDecorator):
+    """Context decorator that disables AutoGraph for the given function/context.
 
-    Args:
-        fn (Callable): the original function object that must not be converted
-
-    Returns:
-        fn: the function not being converted
-
-    **Example**
+    **Example 1: as a function decorator**
 
     .. code-block:: python
 
-        @disable_autograph
+        @disable_autograph()
```

Comment:
It's a shame this is now required 🤔

Reply:
Got an error when I didn't use the parenthesis. Is there a way to avoid this enforcement?

Reply:
So I think it's mainly a question of what kind of object we are making `disable_autograph` to be. Right now it is a class that needs to be instantiated and then called to wrap the function (the `__call__` method is inherited from the `ContextDecorator` base class).

An alternative would be to create a singleton instance, which means we could skip the instantiation at the call site. Something like the following:

```py
class DisableAutograph(...):
    ...

disable_autograph = DisableAutograph()
```

could then be invoked as
```py
@disable_autograph
def f():
```
and
```
with disable_autograph:
    f()
```

Not a huge deal either way though.